daemon/docker.go

```diff
@@ -83,13 +84,23 @@ func deploy(repo *git.Repository, cli *docker.Client) error {
 	// second container to require access to the docker socket.
 	// See https://cloud.google.com/community/tutorials/docker-compose-on-container-optimized-os
 	log.Println("Bringing project online.")
+
+	remotes, err := repo.Remotes()
+	if err != nil {
+		return err
+	}
+
 	ctx := context.Background()
+	url := remotes[0].Config().URLs[0]
+	r, _ := regexp.Compile("(?:/)([^/]+)(?:.git)$")
+	repoName := r.FindStringSubmatch(url)[1]
```

Comment:
If you want to make these lines testable, one way would to be turn it into a helper function in package `common`:

```go
func getRepoNameFromURL(url string) string {
    // ...
}
```

Then you can just call that function alone for tests

Reply:
That sounds good, I'll try that out. 

Is there a way to test manually test this to make sure that a container is successfully created with the appropriate name? I know that the individual `-p` command works as expected for docker-compose, but I wanted to see if it works in the context of the entire method.

Reply:
Yep! Start with this: https://github.com/ubclaunchpad/inertia#testing

Then try doing:
```bash
docker ps
docker exec -it testvps sh
# this will take you into a shell inside the testenv container
docker ps
# should list containers that were started by daemon - these containers should have names prefixed by the project name you extracted
```

Reply:
Thanks! It looks like it's working perfectly.

Reply:
... spoke too soon, it's not working >_<